pkg/gadgets/internal/networktracer/tracer.go

```diff
@@ -117,18 +117,30 @@ func (t *Tracer[Event]) newAttachment(
 }
 
 func NewTracer[Event any](
-	spec *ebpf.CollectionSpec,
 	baseEvent func(ev types.Event) *Event,
 	processEvent func(rawSample []byte, netns uint64) (*Event, error),
```

Comment:
I think these two should be moved to `Listen()` or `Run()` as they are not really needed when creating the network tracer.